sdk/spring/spring-cloud-azure-resourcemanager/src/test/java/com/azure/spring/cloud/resourcemanager/implementation/crud/ServiceBusTopicSubscriptionCrudTests.java

```diff
@@ -0,0 +1,122 @@
+        ManagementException exception = getManagementException(statusCode, message);
+        when(namespaces.getByResourceGroup(resourceMetadata.getResourceGroup(), getKey().getT1()))
+            .thenReturn(namespace);
+
+        Topics topics = mock(Topics.class);
+        Topic topic = mock(Topic.class);
+        ServiceBusSubscriptions serviceBusSubscriptions = mock(ServiceBusSubscriptions.class);
+        when(namespace.topics()).thenReturn(topics);
+        when(topics.getByName(getKey().getT2())).thenReturn(topic);
+        when(topic.subscriptions()).thenReturn(serviceBusSubscriptions);
+        when(serviceBusSubscriptions.getByName(getKey().getT2())).thenThrow(exception);
+    }
+
+    @Override
+    void createStubManagementException() {
+        ServiceBusNamespaces namespaces = mock(ServiceBusNamespaces.class);
+        ServiceBusNamespace namespace = mock(ServiceBusNamespace.class);
+        when(resourceManager.serviceBusNamespaces()).thenReturn(namespaces);
+        ManagementException exception = getManagementException(500, "Create service bus namespace exception");
+        when(namespaces.getByResourceGroup(resourceMetadata.getResourceGroup(), getKey().getT1()))
+            .thenReturn(namespace);
+
+        Topics topics = mock(Topics.class);
+        Topic topic = mock(Topic.class);
+        ServiceBusSubscriptions serviceBusSubscriptions = mock(ServiceBusSubscriptions.class);
+        when(namespace.topics()).thenReturn(topics);
+        when(topics.getByName(getKey().getT2())).thenReturn(topic);
+        when(topic.subscriptions()).thenReturn(serviceBusSubscriptions);
+        when(serviceBusSubscriptions.getByName(getKey().getT3())).thenThrow(exception);
+
+        ServiceBusTopicCrud serviceBusTopicCrud = mock(ServiceBusTopicCrud.class);
+        when(serviceBusTopicCrud.get(Tuples.of(NAMESPACE, TOPIC_NAME))).thenReturn(null);
+
+        ServiceBusSubscription.DefinitionStages.Blank define = mock(ServiceBusSubscription.DefinitionStages.Blank.class);
+        when(serviceBusSubscriptions.define(SUBSCRIPTION_NAME)).thenReturn(define);
+        when(define.create()).thenThrow(exception);
+
+        ServiceBusSubscription.DefinitionStages.WithCreate create = mock(ServiceBusSubscription.DefinitionStages.WithCreate.class);
+        when(create.create()).thenThrow(exception);
+    }
+
+    @Override
+    Tuple3<String, String, String> getKey() {
+        return Tuples.of(NAMESPACE, TOPIC_NAME, SUBSCRIPTION_NAME);
+    }
+
+    @Test
+    void topicDoesNotExistReturnNull() {
+        ServiceBusTopicCrud topicCrud = mock(ServiceBusTopicCrud.class);
+        ServiceBusTopicSubscriptionCrud topicSubCrud = new ServiceBusTopicSubscriptionCrud(this.resourceManager,
+            this.resourceMetadata, topicCrud);
+
+        when(topicCrud.get(Tuples.of(NAMESPACE, TOPIC_NAME))).thenReturn(null);
+
+        ServiceBusSubscription actualGet = topicSubCrud.get(getKey());
+        Assertions.assertNull(actualGet);
+    }
```

Comment:
Do we need a case for when topic is not null?

Reply:
added